setup_wizard/character_rig_setup/rig_script.py

```diff
@@ -9,4 +14,1848 @@ def rig_character(
+            var = influence_driver.variables.new()
+            var.name = "bone"
+            var.type = 'TRANSFORMS'
+            
+            var.targets[0].id = armature
+            var.targets[0].bone_target = bone
+            var.targets[0].transform_space = 'LOCAL_SPACE'
+            var.targets[0].transform_type = trans_rot
+
+            influence_driver.type = 'SCRIPTED'
+            influence_driver.expression = expression 
+
+            depsgraph = bpy.context.evaluated_depsgraph_get()
+            depsgraph.update()
+            # END DRIVER STUFF
+            
+        co.target_space = "LOCAL"
+        co.owner_space = "LOCAL"
+        co.map_from = "ROTATION"
+        co.map_to = "ROTATION"
+        
+        # Do transform constraint math
+        # MAP FROM (put in t/e block?)
+        co.from_min_x_rot = f_min_x
+        co.from_max_x_rot = f_max_x
+        
+        co.from_min_y_rot = f_min_y
+        co.from_max_y_rot = f_max_y
+        
+        co.from_min_z_rot = f_min_z
+        co.from_max_z_rot = f_max_z
+        
+        # MAP TO
+        co.map_to_x_from = map_x
+        co.to_min_x_rot = t_min_x
+        co.to_max_x_rot = t_max_x
+        
+        co.map_to_y_from = map_y
+        co.to_min_y_rot = t_min_y
+        co.to_max_y_rot = t_max_y
+        
+        co.map_to_z_from = map_z
+        co.to_min_z_rot = t_min_z
+        co.to_max_z_rot = t_max_z
+         
+    def leg_match(bone_name):
+        if " R " in bone_name:
+            return "DEF-thigh.R"
+        
+        elif " L " in bone_name:
+            return "DEF-thigh.L"
+
+    allow_back_move = True
+
+    # Interesting thing here. Sometimes, character only have a short "skirt" at the front, that makes no sense in moving back with the leg.
+    def identify_front_skirt():
+        for this_skirt_bone in front_skirt_bones:
+            if this_skirt_bone[-1] == "3":
+                allow_back_move = True
+
+           
+    def identify_front_skirt2():
+        global allow_back_move
+        distance_allowed = 0.03
+        bpy.ops.object.mode_set(mode='EDIT') 
+        dist = []
+        if allow_back_move:
+            for this_skirt_bone in front_skirt_bones:
+                if this_skirt_bone[-1] == "2":
+                    temp2 = this_skirt_bone
+                    temp1 = temp2[:-1] + "1"
+                    try:
+                        distance = abs(armature.edit_bones[temp2].head[1] - armature.edit_bones[temp1].tail[1])
+                        print("distance: " + str(distance))
+                        dist.append(distance)
+                    except:
+                        pass
+            avg = sum(dist)/len(dist)
+            print("avg: " + str(avg))
+            if avg < distance_allowed:
+                allow_back_move = False
+        bpy.ops.object.mode_set(mode='POSE')             
+                    
+                    
+
+
+    #identify_front_skirt2()
+    #identify_front_skirt()
```

Comment:
I don't see these functions being used `identify_front_skirt()`, `identify_front_skirt2()` and `leg_match()`,

I'm fine with leaving them, especially if we think we'll use them later, but can you give some context on what they are used for and when they'd be used? In case we need to refer to them later on.